cmd/restic/cmd_restore.go

```diff
@@ -174,44 +149,28 @@ func runRestore(ctx context.Context, opts RestoreOptions, gopts GlobalOptions,
 		msg.E("Warning: %s\n", message)
 	}
 
-	excludePatterns := filter.ParsePatterns(opts.Exclude)
-	insensitiveExcludePatterns := filter.ParsePatterns(opts.InsensitiveExclude)
 	selectExcludeFilter := func(item string, _ string, node *restic.Node) (selectedForRestore bool, childMayBeSelected bool) {
-		matched, err := filter.List(excludePatterns, item)
-		if err != nil {
-			msg.E("error for exclude pattern: %v", err)
-		}
-
-		matchedInsensitive, err := filter.List(insensitiveExcludePatterns, strings.ToLower(item))
-		if err != nil {
-			msg.E("error for iexclude pattern: %v", err)
-		}
+		for _, rejectFn := range excludePatternFns {
+			matched := rejectFn(item)
```

Comment:
That logic is partially broken. If any `rejectFn` returns true then the file must be excluded. Right now, the loop only ever looks at the first function in `excludePatternFns`

You have to combine the results as follows:
```go
matched := false
for _, rejectFn := range excludePatternFns {
    matched = matched || rejectFn(item)
}
```

Reply:
My thought process was that I must exit the first match I get. Because it's an || operation. But I introduced an issue. I have resolved this.

Reply:
Let me know if that makes sense if we break after the first pattern it matches, it will be much faster. We don't have to do redundant pattern matching with other patterns.

Reply:
A short-circuit semantic sounds reasonable. That also works for the include filter, although there the optimization is only possible if both `selectedForRestore` and `childMayBeSelected` are true within the loop.

Reply:
@MichaelEischer I have made the changes, please check if it is correct.